common/beerocks/bwl/dwpal/ap_wlan_hal_dwpal.cpp

```diff
@@ -1868,6 +1872,11 @@ bool ap_wlan_hal_dwpal::process_dwpal_event(char *buffer, int bufLen, const std:
         msg->params.vap_id       = beerocks::utils::get_ids_from_iface_string(VAP).vap_id;
         msg->params.mac          = beerocks::net::network_utils::mac_from_string(MACAddress);
         msg->params.capabilities = {};
+        auto assoc_frame_len     = strnlen(assoc_req, ASSOCIATION_FRAME_SIZE);
+        auto sub_str             = std::string(&assoc_req[56], assoc_frame_len - 56 - 18);
+        msg->params.association_frame_length = sub_str.length();
+
+        std::copy_n(&sub_str[0], sub_str.length() + 1, msg->params.association_frame);
```

Comment:
Again, it should already have been converted to binary here.

I realize I said before that you could choose to pass it around as string internally, but really there is no reason to do that, so let's just do the right thing and convert to binary here.

Reply:
done.